lib/Fmt_ast.ml

```diff
@@ -2914,26 +2906,46 @@ and fmt_expression c ?(box = true) ?(pro = noop) ?eol ?parens
       pro $ fmt_indexop_access c ctx ~fmt_atrs ~has_attr ~parens x
   | Pexp_hole -> pro $ hvbox 0 (fmt_hole () $ fmt_atrs)
   | Pexp_beginend e ->
-      fmt_beginend c ~box ~pro ~ctx ~fmt_atrs ~ext ~indent_wrap ?eol e
+      fmt_beginend c ~loc:pexp_loc ~box ~pro ~ctx ~fmt_atrs ~ext ~indent_wrap
+        ?eol e
   | Pexp_parens e ->
       pro
       $ hvbox 0
           (fmt_expression c ~box ?eol ~parens:true ~indent_wrap ?ext
              (sub_exp ~ctx e) )
       $ fmt_atrs
 
-and fmt_beginend c ?(box = true) ?(pro = noop) ~ctx ~fmt_atrs ~ext
+and fmt_lazy c ~ctx ?(pro = noop) ~fmt_atrs ~ext ~parens e =
+  let lazy_ = str "lazy" $ fmt_extension_suffix c ext in
+  match e with
+  | {pexp_desc= Pexp_beginend _; _} as e ->
+      pro
+      $ hvbox 2
+          (Params.Exp.wrap c.conf ~parens
+             ( fmt_expression c ~pro:(lazy_ $ str " ") (sub_exp ~ctx e)
```

Comment:
It would be nice to add a comment explaining that `str " "` plays a role in formatting `lazy begin fun .... ->`. For `lazy begin match` of `lazy begin fun` with small argument list, `space_break` should be fine.

Reply:
Space break is indeed fine, I just need to display the `pro` in the same box as the `begin` keyword.
I dont think `lazy begin fun` with a large list of arguments is a big concern.

Reply:
Alright, I tested `lazy begin fun`, and its indeed pretty ugly, so I restored to `str " "`, and added a comment.